packages/data/src/plugins/actions.js

```diff
@@ -125,6 +126,62 @@ export function* installAndActivatePlugins( plugins ) {
 	}
 }
 
+export const createErrorNotice = ( errorMessage ) => {
+	return dispatch( 'core/notices', 'createNotice', errorMessage );
+};
+
+export function* installJetpackAndConnect() {
+	try {
+		yield dispatch( STORE_NAME, 'installPlugins', [ 'jetpack' ] );
+		yield dispatch( STORE_NAME, 'activatePlugins', [ 'jetpack' ] );
+
+		const url = yield select( STORE_NAME, 'getJetpackConnectUrl', {
+			redirect_url: getAdminLink( 'admin.php?page=wc-admin' ),
+		} );
+
+		// getJetpackConnectUrl doesn't throw errors, it sets error state if
+		// something went wrong.
+		const error = yield select(
+			STORE_NAME,
+			'getPluginsError',
+			'getJetpackConnectUrl'
+		);
+
+		if ( error ) {
+			yield createErrorNotice( error );
+		} else {
+			window.location = url;
+		}
+	} catch ( error ) {
+		yield createErrorNotice( error.message );
+	}
+}
+
+export function* connectToJetpack( failureRedirect ) {
+	try {
+		const url = yield select( STORE_NAME, 'getJetpackConnectUrl', {
+			redirect_url: getAdminLink( 'admin.php?page=wc-admin' ),
+		} );
+
+		// getJetpackConnectUrl doesn't throw errors, it sets error state if
+		// something went wrong.
+		const error = yield select(
+			STORE_NAME,
+			'getPluginsError',
+			'getJetpackConnectUrl'
+		);
+
+		if ( error ) {
+			yield createErrorNotice( error );
+			window.location = failureRedirect;
```

Comment:
This will potentially give us a hard redirect as opposed to using the router `getHistory().push()`.  `window.location` is probably more universally useful and makes sense, but just wanted to get your thoughts before merging.

Reply:
@joshuatf yes I think the hard redirect will support *all* failure scenarios for outside consumers. and in rare cases maybe us as well, such as where we host our components in pages other than `admin.php`. We could also make this configurable in future if such a feature is desired?